openslides_backend/action/actions/motion/json_upload.py

```diff
@@ -0,0 +1,723 @@
+    submitter_lookup: Dict[str, List[Dict[str, Any]]] = {}
+    supporter_lookup: Dict[str, List[Dict[str, Any]]] = {}
+    category_lookup: Dict[str, List[Dict[str, Any]]] = {}
+    tags_lookup: Dict[str, List[Dict[str, Any]]] = {}
+    block_lookup: Dict[str, List[Dict[str, Any]]] = {}
+    _first_state_id: Optional[int] = None
+    _operator_username: Optional[str] = None
+
+    _last_motion_mock_id: Optional[int] = None
+    _user_ids_to_meeting_user: Dict[int, Any]
+
+    def update_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
+        # transform instance into a correct create/update payload
+        # try to find a pre-existing motion with the same number
+        # if there is one, validate for a motion.create, otherwise for a motion.update
+        # using get_create_payload_integrity_error_message and get_update_payload_integrity_error_message
+
+        data = instance.pop("data")
+        data = self.add_payload_index_to_action_data(data)
+        self.setup_lookups(data, instance["meeting_id"])
+
+        # enrich data with meeting_id
+        for entry in data:
+            entry["meeting_id"] = instance["meeting_id"]
+
+        self.rows = [self.validate_entry(entry) for entry in data]
+
+        # generate statistics
+        itemCount = len(self.rows)
+        state_to_count = {state: 0 for state in ImportState}
+        for row in self.rows:
+            state_to_count[row["state"]] += 1
+            state_to_count[ImportState.WARNING] += self.count_warnings_in_payload(
+                row.get("data", {}).values()
+            )
+            row["data"].pop("payload_index", None)
+
+        self.statistics = [
+            {"name": "total", "value": itemCount},
+            {"name": "created", "value": state_to_count[ImportState.NEW]},
+            {"name": "updated", "value": state_to_count[ImportState.DONE]},
+            {"name": "error", "value": state_to_count[ImportState.ERROR]},
+            {"name": "warning", "value": state_to_count[ImportState.WARNING]},
+        ]
+
+        self.set_state(
+            state_to_count[ImportState.ERROR], state_to_count[ImportState.WARNING]
+        )
+        self.store_rows_in_the_import_preview("motion")
+        return {}
+
+    def validate_entry(self, entry: Dict[str, Any]) -> Dict[str, Any]:
+        messages: List[str] = []
+        id_: Optional[int] = None
+        meeting_id: int = entry["meeting_id"]
+        set_entry_id = False
+
+        if (is_amendment := entry.get("motion_amendment")) and isinstance(
+            is_amendment, bool
+        ):
+            entry["motion_amendment"] = {
+                "value": is_amendment,
+                "info": ImportState.WARNING,
+            }
+            messages.append("Amendments cannot be correctly imported")
+
+        if (category_name := entry.get("category_name")) and isinstance(
+            category_name, str
+        ):
+            category_prefix = entry.get("category_prefix") or None
```

Comment:
You don't need   the `or None`. The get-method returns the value, if the dictionary has the attribute or it returns None.

Reply:
I need `or None` because I need this to ignore empty strings

Reply:
Then please write a test for this edge case:-).

Reply:
Already exists

Reply:
If the `and None` is necessary, there should be a reason why it is necessary. I would expect removing it, changes the behaviour. And this change should be manifested in a failing test

Reply:
I removed the or None and it still works, seems it was unnecessary